### ISSUE TITLE
`almost_swapped` false positive on `#[derive(clap::Subcommand)]`

### ISSUE THREAD

**Issue by 9999years** [rust-lang/rust-clippy#10676]
### Summary

`clap` 4.0 raises what looks like a spurious `almost_swapped` lint.

Frustratingly, I don't appear to be able to silence the lint with `#![allow(clippy::almost_swapped)]`.

### Lint Name

almost_swapped

### Reproducer

I tried this code:

```rust
#[derive(Debug, Clone, clap::Subcommand)]
pub enum Command {
    /// Uninstall Nix and exit when finished.
    UninstallNix,
}
```

I saw this happen:

```ShellSession
$ cargo clippy
error: this looks like you are trying to swap `__clap_subcommand` and `clap::Subcommand`
   --> src/cli.rs:117:24
    |
117 | #[derive(Debug, Clone, clap::Subcommand)]
    |                        ^^^^^^^^^^^^^^^^
    |
    = note: or maybe you should use `std::mem::replace`?
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#almost_swapped
note: the lint level is defined here
   --> src/cli.rs:117:24
    |
117 | #[derive(Debug, Clone, clap::Subcommand)]
    |                        ^^^^^^^^^^^^^^^^
    = note: `#[deny(clippy::almost_swapped)]` implied by `#[deny(clippy::correctness)]`
    = note: this error originates in the derive macro `clap::Subcommand` (in Nightly builds, run with -Z macro-backtrace for more info)
```

I expected to see this happen:

```ShellSession
$ cargo clippy
All good :)
```

### Version

```text
rustc 1.69.0 (84c898d65 2023-04-16)
binary: rustc
commit-hash: 84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc
commit-date: 2023-04-16
host: aarch64-apple-darwin
release: 1.69.0
LLVM version: 15.0.7
```

**Comment by 9999years** [comment]
This is a duplicate of https://github.com/rust-lang/rust-clippy/issues/10421, but even though that issue is closed this occurs with today's stable release.

**Comment by 9999years** [comment]
Updating from `clap v4.0.18 -> v4.2.4` fixed the issue for me. Still seems weird but closing.

**Comment by shinypb** [comment]
@9999years I'm seeing this too! We're intentionally all the way back on clap 3.2.23 for various reasons; would you be open to re-opening this issue even though there's a workaround? 🙏
